src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -219,18 +222,90 @@ def q_impl(
 
         p = weights_q_values.shape[0]
 
+        def enc_mul(x, y):
+            """Encrypted multiplication trick.
+
+            Args:
+                x (numpy.ndarray): first input numpy array
+                y (numpy.ndarray): second input numpy array
+            """
+            with tag("trick"):
+                add = x + y
```

Comment:
this won't broadcast when shapes match no? it feels like running on a (N,N) x (N,N) matrix will not work

Reply:
I will add (N, N) tests. But it should work yes.